src/status_im2/contexts/chat/messages/list/view.cljs

```diff
@@ -102,47 +105,107 @@
+              {:transform [{:translate-y y}]}
+              {:bottom   (+ composer.constants/composer-default-height (:bottom insets) 6)
+               :position :absolute
+               :left     0
+               :right    0})}
      [quo/floating-shell-button
       (merge {:jump-to
               {:on-press #(do
                             (rf/dispatch [:chat/close true])
                             (rf/dispatch [:shell/navigate-to-jump-to]))
-               :label    (i18n/label :t/jump-to)}}
+               :label    (i18n/label :t/jump-to)
+               :style    {:align-self :center}}}
              (when @show-floating-scroll-down-button
                {:scroll-to-bottom {:on-press scroll-to-bottom}}))
-      {:position :absolute
-       :bottom   6}]]))
+      {}]]))
+
+
+(defn messages-list-content
+  [{:keys [chat-id] :as chat} insets keyboard-shown]
+  (fn []
+    (let [context  (rf/sub [:chats/current-chat-message-list-view-context])
+          messages (rf/sub [:chats/raw-chat-messages-stream chat-id])]
+      [rn/view
+       {:style {:flex 1}}
+       ;; NOTE: DO NOT use anonymous functions for handlers
+       [rn/flat-list
+        {:key-fn                       list-key-fn
+         :ref                          list-ref
+         :header                       [list-header chat]
+         :footer                       [list-footer chat]
+         :data                         messages
+         :render-data                  {:context        context
+                                        :keyboard-shown keyboard-shown}
+         :render-fn                    render-fn
+         :on-viewable-items-changed    on-viewable-items-changed
+         :on-end-reached               list-on-end-reached
+         :on-scroll-to-index-failed    identity ; don't remove this
+         :content-container-style      {:padding-top    (+ composer.constants/composer-default-height
+                                                           (:bottom insets)
+                                                           32)
+                                        :padding-bottom 16}
+         :scroll-indicator-insets      {:top (+ composer.constants/composer-default-height
+                                                (:bottom insets))}
+         :keyboard-dismiss-mode        :interactive
+         :keyboard-should-persist-taps :handled
+         :on-momentum-scroll-begin     state/start-scrolling
+         :on-momentum-scroll-end       state/stop-scrolling
+         :scroll-event-throttle        16
+         :on-scroll                    on-scroll
+         ;; TODO https://github.com/facebook/react-native/issues/30034
+         :inverted                     (when platform/ios? true)
+         :style                        (when platform/android? {:scaleY -1})
+         :on-layout                    on-messages-view-layout}]
+       [:f> shell-button insets]])))
+
+
+(defn use-keyboard-visibility
```

Comment:
could you pls add a comment that this should be replaced by keyboard-hook and why it doesn't work now